typescript/scripts/build.sh

```diff
@@ -13,5 +13,13 @@ lang_root_dirpath="$(dirname "${script_dirpath}")"
 # # ==================================================================================================
 # #                                             Main Logic
 # # ==================================================================================================
-find_results="$(find ${lang_root_dirpath}/lib -name "*.ts")"
-tsc ${find_results} --outFile ${lang_root_dirpath}/build/output.js --module system --moduleResolution node --target es2015
+#find_results="$(find ${lang_root_dirpath} -name "*.ts" -not -path "${lang_root_dirpath}/kurtosis_core_rpc_api_bindings/*" -not -path "${lang_root_dirpath}/node_modules/*")"
+
+tsc ${lang_root_dirpath}/index.ts --outDir ${lang_root_dirpath}/build/ --downlevelIteration true --declaration true
+cp -R ${lang_root_dirpath}/kurtosis_core_rpc_api_bindings/ ${lang_root_dirpath}/build/kurtosis_core_rpc_api_bindings/ #TODO - error checking but after confirming if this is okay solution
```

Comment:
Ooooooo very interesting that this works - so basically we're recreating the directory structure for index.js it looks like. I think this is fine; maybe there's a cleaner way, but whatever.

Only recommendation is DRY, and do`cp -R ${lang_root_dirpath}/kurtosis_core_rpc_api_bindings ${lang_root_dirpath}/` instead. The way you did it is fine, but is vulnerable to typos/changes in the `kurtosis_core_rpc_api_bindings` directory.

The way to think about this is, when you see a trailing `/`, it means "INSIDE of this directory". When you don't see "/", it means "the file/dir itself".

This is also why, if you're moving a bunch of files inside a you expect to exist, MAKE SURE YOU ADD THE TRAILING SLASH! I learned this the hard way in college: I did `mv * typod-directory-name` .... which, because `typod-directory-name` didn't exist, translated to `mv file1 file2 file3 typod-directory-name`.... which meant I ended up with a single file called `typod-directory-name` that contained only the contents of file3. Lost a looooooot of work that day.

Instead, if you do `mv * typod-directory-name/`, the trailing slash tells the shell to put the contents INSIDE the directory. Since `typod-directory-name` doesn't exist and isn't a directory, you'll instead get an error.

Reply:
Also: can you add a quick comment explaining why this step is necessary

Reply:
This was the code I added to typescript/scripts/build.sh

![image](https://user-images.githubusercontent.com/52847195/130083711-d7e345d7-04bf-4c6c-b5bd-f55dad630029.png)

Compiled index.ts into .js and .d.ts, which then took all of the dependencies and compiled them into their own .js and .d.ts files

And then to get rid of the dependency issue of the typescript generated bindings, I copied those over to the hidden build/ directory where index.js can be found

This got it working when I was locally testing if another repo could register what ContainerRunConfigBuilder.entrypointOverrideArgs is (should be null) which it got correct


![image](https://user-images.githubusercontent.com/52847195/130083778-60900190-9e19-4abc-993d-6eec8adf3582.png)


Here is what test.ts looked like


![image](https://user-images.githubusercontent.com/52847195/130083827-f66a716c-75b7-4b36-b457-46f29e69e3ee.png)